File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public static String solution(String[] participant, String[] completion) {
    String answer = "";

    Map<String, Integer> hm = new HashMap<>();
	
    for (String p : participant) {
        hm.put(p, hm.getOrDefault(p, 0)+1);
    }

    for (String c : completion) {
        hm.put(c, hm.get(c) - 1);
    }

    for (String key : hm.keySet()) {
        if (hm.get(key) != 0) {
            answer = key;
            break;
        }
    }

    return answer;
}
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer;
    let hashMap = new Map()

    participant.forEach(e => {
        if(hashMap.has(e)) {
            hashMap.set(e, hashMap.get(e)+1);
        } else {
            hashMap.set(e, 1)    
        }
    })

    // 3)
    completion.forEach(e => {
        if(hashMap.has(e)) {
            hashMap.set(e, hashMap.get(e)-1);
        }
    })

    for(let [key, value] of hashMap) {
        if(value == 1) {
            answer = key
        }
    }

    return answer;
}